src/EFCore.PG.Trgm/Infrastructure/Internal/NpgsqlTrgmOptionsExtension.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.Extensions.DependencyInjection;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Query;
+
+using Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Internal;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Infrastructure.Internal
+{
+    public class NpgsqlTrgmOptionsExtension : IDbContextOptionsExtension
+    {
+        public DbContextOptionsExtensionInfo Info => new ExtensionInfo(this);
+
+        public virtual void ApplyServices(IServiceCollection services)
+        {
+            services.AddEntityFrameworkNpgsqlTrgm();
+        }
+
+        public virtual void Validate(IDbContextOptions options)
+        {
+            var internalServiceProvider = options.FindExtension<CoreOptionsExtension>()?.InternalServiceProvider;
+            if (internalServiceProvider != null)
+            {
+                using (var scope = internalServiceProvider.CreateScope())
+                {
+                    if (scope.ServiceProvider.GetService<IEnumerable<IMethodCallTranslatorPlugin>>()
+                            ?.Any(s => s is NpgsqlTrgmMethodCallTranslatorPlugin) != true)
+                    {
+                        throw new InvalidOperationException($"{nameof(NpgsqlTrgmDbContextOptionsBuilderExtensions.UseTrgm)} requires {nameof(NpgsqlTrgmServiceCollectionExtensions.AddEntityFrameworkNpgsqlTrgm)} to be called on the internal service provider used.");
+                    }
+                }
+            }
+        }
+
+        sealed class ExtensionInfo : DbContextOptionsExtensionInfo
+        {
+            public ExtensionInfo(IDbContextOptionsExtension extension)
+                : base(extension)
+            {
+            }
+
+            public override bool IsDatabaseProvider => false;
+
+            public override long GetServiceProviderHashCode() => 0;
+
+            public override void PopulateDebugInfo(IDictionary<string, string> debugInfo)
+            => debugInfo["Npgsql:" + nameof(NpgsqlTrgmDbContextOptionsBuilderExtensions.UseTrgm)] = "1";
```

Comment:
nit: Let's put the arrow on the previous line and correctly indent the statement.

Reply:
24ff45d1edc00f0f4748be34ee18e0c0eee25f3e fixes this.

Reply:
3b491f43397f0cfc979e12f49a552e94793fa845 puts the arrow on the next line.